### ISSUE TITLE
clock_gettime on macOS

### ISSUE THREAD

**Issue by mehdideveloper** [BatchDrake/suscan#37]
Hey
clock_gettime is not implemented on macOS (neither its related constants )
Look at this for example: https://stackoverflow.com/questions/5167269/clock-gettime-alternative-in-mac-os-x

**Comment by mehdideveloper** [comment]
And I'm not sure what you think about this:

https://github.com/envelope-project/laik/commit/86c180e1c16fffa0761a785dad315a7d524e053b

**Comment by BatchDrake** [comment]
We've been using `clock_gettime` in MacOS for a whie now, what does not exist is `CLOCK_MONOTONIC_COARSE`. We may need to fallback to `CLOCK_MONOTONIC`

**Comment by BatchDrake** [comment]
Also, maybe it is a good idea to rely on Mach functions directly, as described in https://opensource.apple.com/source/Libc/Libc-1158.1.2/gen/clock_gettime.c.auto.html

**Comment by BatchDrake** [comment]
Also, is `mach_boottime_usec` documented anywhere?

**Comment by mehdideveloper** [comment]
Ok. I was wrong. It was not available before OS X 10.12.
I checked `time.h` and it actually is there.
Regarding `CLOCK_MONOTONIC_COARSE`: 
I searched my `time.h` and this is what I have:
```
 _CLOCK_MONOTONIC __CLOCK_AVAILABILITY = 6,
#define CLOCK_MONOTONIC _CLOCK_MONOTONIC
#if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
_CLOCK_MONOTONIC_RAW __CLOCK_AVAILABILITY = 4,
#define CLOCK_MONOTONIC_RAW _CLOCK_MONOTONIC_RAW
_CLOCK_MONOTONIC_RAW_APPROX __CLOCK_AVAILABILITY = 5,
#define CLOCK_MONOTONIC_RAW_APPROX _CLOCK_MONOTONIC_RAW_APPROX
```

**Comment by mehdideveloper** [comment]
Where does `mach_boottime_usec` come from? I can't find it in any files in my `include`

**Comment by sudeepan** [comment]
Is there a `mach` equivalent of `CLOCK_MONOTONIC_COARSE` ?

**Comment by sudeepan** [comment]
> Ok. I was wrong. It was not available before OS X 10.12.
> I checked `time.h` and it actually is there.
> Regarding `CLOCK_MONOTONIC_COARSE`: 
> I searched my `time.h` and this is what I have:
> ```
>  _CLOCK_MONOTONIC __CLOCK_AVAILABILITY = 6,
> #define CLOCK_MONOTONIC _CLOCK_MONOTONIC
> #if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
> _CLOCK_MONOTONIC_RAW __CLOCK_AVAILABILITY = 4,
> #define CLOCK_MONOTONIC_RAW _CLOCK_MONOTONIC_RAW
> _CLOCK_MONOTONIC_RAW_APPROX __CLOCK_AVAILABILITY = 5,
> #define CLOCK_MONOTONIC_RAW_APPROX _CLOCK_MONOTONIC_RAW_APPROX
> ```
> 

Is there a `mach` equivalent of `CLOCK_MONOTONIC_COARSE` ?

**Comment by sudeepan** [comment]
> We've been using `clock_gettime` in MacOS for a whie now, what does not exist is `CLOCK_MONOTONIC_COARSE`. We may need to fallback to `CLOCK_MONOTONIC`

Is there a `mach` equivalent of `CLOCK_MONOTONIC_COARSE` ?
